packages/compile-solidity/run.js

```diff
@@ -14,11 +14,18 @@ async function run(rawSources, options, language = "Solidity") {
 
   // Ensure sources have operating system independent paths
   // i.e., convert backslashes to forward slashes; things like C: are left intact.
+  // we also strip the project root (to avoid it appearing in metadata)
+  // and replace it with "project://"
   const {
     sources,
     targets,
     originalSourcePaths
-  } = Common.Sources.collectSources(rawSources, options.compilationTargets);
+  } = Common.Sources.collectSources(
+    rawSources,
+    options.compilationTargets,
+    options.working_directory,
+    "project:/" //only one slash, other will already be present
```

Comment:
Maybe this is a little pedantic but is it technically more proper to have this argument be the whole string? Then we should strip off the relative path portion and replace it with "exactly" what we will be using? Like what if (and yeah, I think this is pretty pedantic) one day we decide to use "project::" instead of "project:/"?

Reply:
agree about replacement style - better to replace the whole thing

Reply:
+1 on that as well

Reply:
OK, I've changed this.